src/collection.js

```diff
@@ -1047,33 +1045,40 @@ export default class Collection {
         new Error(`Server is asking clients to back off; retry in ${seconds}s or use the ignoreBackoff option.`));
     }
 
-    this._lastModified = await this.db.getLastModified();
-
     const client = this.api.bucket(options.bucket || this.bucket).collection(options.collection || this.name);
-    let syncResult;
+
+    const result = new SyncResultObject();
     try {
-      const result = new SyncResultObject();
-      const pullResult = await this.pullChanges(client, result, options);
-      const pushResult = await this.pushChanges(client, pullResult, options);
-      // Avoid performing a last pull if nothing has been published.
-      if (pushResult.published.length === 0) {
-        syncResult = pushResult;
-      } else {
+      // Fetch last changes from the server.
+      await this.pullChanges(client, result, options);
```

Comment:
@leplatrem explained to me that although we used to have this sort of return-value chaining, which suggests immutability, things were also being modified in-place, so he took out the "fake" return values.